### PR TITLE
feat: add `flow_id` and `drop_if_exists` to `DropFlowExpr`

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

add `flow_id` and `drop_if_exists` to `DropFlowExpr`

## Checklist

- [ ]  I have written the necessary comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: proto/greptime/v1/flow/server.proto [hunk 1]
```diff
@@ -52,15 +52,15 @@ message FlowResponse {
   uint64 affected_rows = 2;
   map<string, bytes> extension = 3;
   // affected task ids
-  repeated TaskId affected_tasks = 4;
+  repeated FlowId affected_tasks = 4;
```
Comment (fengjiachun) [inline proto/greptime/v1/flow/server.proto:55]:
```suggestion
  repeated FlowId affected_flows = 4;
```

Comment (fengjiachun) [reply proto/greptime/v1/flow/server.proto:55]: Can we put the 'extensions' field at the end?

File: proto/greptime/v1/flow/server.proto [hunk 2]
```diff
@@ -52,15 +52,15 @@ message FlowResponse {
   uint64 affected_rows = 2;
   map<string, bytes> extension = 3;
   // affected task ids
```
Comment (fengjiachun) [inline proto/greptime/v1/flow/server.proto:54]:
```suggestion
  // affected flow ids
```

**Review by tisonkun — APPROVED** [review]

File: proto/greptime/v1/flow/server.proto [hunk 3]
```diff
@@ -59,17 +59,17 @@ message FlowRequest {
 message FlowResponse {
   ResponseHeader header = 1;
   uint64 affected_rows = 2;
-  map<string, bytes> extension = 3;
-  // affected task ids
-  repeated TaskId affected_tasks = 4;
+  // affected flow ids
+  repeated FlowId affected_flows = 3;
+  map<string, bytes> extension = 4;
```
Comment (evenyag) [inline proto/greptime/v1/flow/server.proto:64]:
```suggestion
  map<string, bytes> extensions = 4;
```

**Review by discord9 — APPROVED** [review]

**Pull request merged**